src/core.c/CompUnit/Repository/FileSystem.pm6

```diff
@@ -220,6 +220,44 @@ class CompUnit::Repository::FileSystem
 
         return ($distribution,);
     }
+    multi method candidates(Str:D :$file!, Str:D :$name!, :$auth, :$ver, :$api) {
```

Comment:
This api feels a bit weird. From a purity standpoint this method should have just been `method candidate(CompUnit::DependencySpecification: $spec)` - the other candidates were added for the convenience of not having to type out the `CompUnit::DependencySpecification` part. i.e. one type of parameters are meant to map directly to the other. In our CUR code `:$file` is sometimes is treated as the short-name, and other times (like when `:$name!`) it is not. It sort of seems like maybe `CompUnit::DependencySpecification` should have an attribute for a file path. I'm not sure if that is a great idea or not, but something like `lib/Foo/Bar.rakumod` does sort of feel like a type of exact dependency specification.